tests/bsim/bluetooth/mesh/src/test_cdp1.c

```diff
@@ -0,0 +1,433 @@
+	test_p1_cfg_srv_mod, test_p1_cfg_cli_mod, test_p1_mod1,
+	test_p1_mod2,        test_p1_mod3,        test_p1_vnd1,
+};
+
+static const struct bt_mesh_comp_p1_model_item test_p1_elem1_models[] = {
+	test_p1_mod4,
+};
+
+static const struct bt_mesh_comp_p1_model_item test_p1_elem2_models[] = {
+	test_p1_mod5,
+};
+
+static const struct bt_mesh_comp_p1_model_item test_p1_elem3_models[] = {
+	test_p1_mod5,
+};
+
+static const struct bt_mesh_comp_p1_model_item test_p1_elem4_models[] = {
+	test_p1_mod6,
+};
+
+static const struct bt_mesh_comp_p1_model_item *test_p1_elem_models[] = {
+	test_p1_elem0_models, test_p1_elem1_models, test_p1_elem2_models,
+	test_p1_elem3_models, test_p1_elem4_models,
+};
+
+static const struct bt_mesh_comp_p1_elem test_p1_elem0 = {
+	.nsig = 5,
+	.nvnd = 1,
+};
+
+static const struct bt_mesh_comp_p1_elem test_p1_elem1 = {
+	.nsig = 1,
+	.nvnd = 0,
+};
+
+static const struct bt_mesh_comp_p1_elem test_p1_elem2 = {
+	.nsig = 1,
+	.nvnd = 0,
+};
+
+static const struct bt_mesh_comp_p1_elem test_p1_elem3 = {
+	.nsig = 1,
+	.nvnd = 0,
+};
+
+static const struct bt_mesh_comp_p1_elem test_p1_elem4 = {
+	.nsig = 1,
+	.nvnd = 0,
+};
+
+static struct bt_mesh_comp_p1_elem test_p1_elems[] = {
+	test_p1_elem0, test_p1_elem1, test_p1_elem2, test_p1_elem3, test_p1_elem4,
+};
+
+static void provision_and_configure(struct bt_mesh_test_cfg cfg)
+{
+	int err;
+	uint8_t status;
+
+	err = bt_mesh_provision(net_key, 0, 0, 0, cfg.addr, cfg.dev_key);
+	if (err) {
+		FAIL("Provisioning failed (err %d)", err);
+	}
+
+	err = bt_mesh_cfg_cli_app_key_add(0, cfg.addr, 0, 0, app_key, &status);
+	if (err || status) {
+		FAIL("AppKey add failed (err %d, status %u)", err, status);
+	}
+}
+
+static void verify_model_item(struct bt_mesh_comp_p1_model_item *mod_item, int elem_idx,
+			      int mod_idx, int offset)
+{
+	ASSERT_EQUAL(test_p1_elem_models[elem_idx][mod_idx + offset].cor_present,
+		     mod_item->cor_present);
+	ASSERT_EQUAL(test_p1_elem_models[elem_idx][mod_idx + offset].format, mod_item->format);
+	ASSERT_EQUAL(test_p1_elem_models[elem_idx][mod_idx + offset].ext_item_cnt,
+		     mod_item->ext_item_cnt);
+	if (mod_item->cor_present) {
+		ASSERT_EQUAL(test_p1_elem_models[elem_idx][mod_idx + offset].cor_id,
+			     mod_item->cor_id);
+	}
+}
+
+static void verify_ext_item(struct bt_mesh_comp_p1_ext_item *ext_item, int elem_idx, int mod_idx,
+			    int offset)
+{
+	struct bt_mesh_comp_p1_ext_item *test_p1_ext_item;
+
+	switch (elem_idx*100 + (mod_idx + offset)) {
+	case 3:	/* elem_idx=0, mod_idx=3, offset = 0*/
+		test_p1_ext_item = &test_p1_ext_mod2_mod1;
+		break;
+	case 4: /* elem_idx=0, mod_idx=4, offset = 0*/
+		test_p1_ext_item = &test_p1_ext_mod3_mod4;
+		break;
+	case 5: /* elem_idx=0, mod_idx=0, offset = 5*/
+		test_p1_ext_item = &test_p1_ext_vnd1_mod1;
+		break;
+	case 400: /* elem_idx=4, mod_idx=0*/
+		test_p1_ext_item = &test_p1_ext_mod6_mod1;
+		break;
+	default:
+		return;
```

Comment:
Shouldn't be `FAIL` here instead of `return`?

Reply:
Agreed, changed to `FAIL`.